artiq/devices/novatech409B/novatech409B_client.py

```diff
@@ -0,0 +1,131 @@
+    parser.add_argument("--echo", type=str, action="store", nargs=1,
+        help="request that server echo back a quoted string")
+    parser.add_argument("--verbose", action="store_true",
+        help="increase output verbosity")
+
+    subparsers = parser.add_subparsers(dest="subparser_name")
+
+    restricted_float_freq = partial(restricted_float, 0.0, 171.1276031)
+    parser_setfreq = subparsers.add_parser("freq",
+                                           help="set frequency")
+    parser_setfreq.add_argument("f", type=restricted_float_freq,
+                                help="frequency in MHz"
+                                "[0.0,171.1276031]")
+    parser_setfreq.add_argument("--channel", default=-1, type=int,
+                        choices=range(0, 4),
+                        help="which channel to set; default is ALL")
+
+    parser_setphase = subparsers.add_parser("phase",
+                                            help="set phase")
+    parser_setphase.add_argument(
+        "p",type=partial(restricted_float, 0.0, 1.0),
+        nargs=4,
+        help="p0 p1 p2 p3 are phases for all four "
+        "channels in cycles (1=360 deg)")
+
+    parser_gain=subparsers.add_parser("gain", help="set output gain")
+    parser_gain.add_argument("g",
+                type=partial(restricted_float, 0.0, 1.0),
+                help="waveform amplitude [0.0, 1.0]")
+    parser_gain.add_argument("--channel",
+                        default=-1, type = int,
+                        choices=range(0, 4),
+                        help="which channel to set; default is ALL")
+
+    parser_sweep_freq = subparsers.add_parser("sweep-freq",
+                                              help="sweep frequency")
+    parser_sweep_freq.add_argument("p0", type=restricted_float_freq,
+                                    help="starting freq in MHz")
+    parser_sweep_freq.add_argument("p1", type=restricted_float_freq,
+                                    help="ending freq in MHz")
+    parser_sweep_freq.add_argument("t", type=float,
+                                    help="sweep time in sec")
+    parser_sweep_freq.add_argument("--channel",
+                        default=-1, type=int,
+                        choices=range(0, 4),
+                        help="which channel to set; default is ALL")
+
+    parser_reset = subparsers.add_parser("reset", help="reset device")
+    parser_reset.add_argument("reset",
+                help="reset device", action="store_true")
+    parser_eeprom = subparsers.add_parser("save-to-eeprom",
+                                          help="save to EEPROM")
+    parser_eeprom.add_argument("save-to-eeprom",
+                        action="store_true",
+                        help="saves current state into EEPROM "
+                        "and sets valid flag; state used as default"
+                        " upon next power up or reset")
+    return parser
+
+
+def _get_args():
```

Comment:
``define_parser`` is called ``get_argparser`` elsewhere in ARTIQ. Having it as a separate function enables sphinx to extract the argument info.